### PR TITLE
fix: Remove navigation error with external surface and no layers

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
Solve the issue show in issue #636.

Instead of obtaining the layer id from the layer obtain it from the first surface. Should be exactly the same think as all the surface visited in targetSurfaces should be a the same layer (if they are on no layer, the id will be 0 and everything is fine).

I also simplified the way the surface are compared to the external surface avoiding a copy of the vector.

I couldn't reproduce the Error in #636 since I don't have Pythia on my computer but I am a bit surprise that a situation with surfaces and no layer can be reached (I couldn't reproduce it with the propagation example alone).

### CODE REVIEW

**Comment by paulgessinger** [comment]
Would you be able to reproduce this error state in a unit test? This seems to be one of the edge cases that would benefit from being covered.

**Comment by Corentin-Allaire** [comment]
> Would you be able to reproduce this error state in a unit test? This seems to be one of the edge cases that would benefit from being covered.

To be honest I am not completely sure how we reach that edge case to begin with... If I can find a bit of during the next few days I will try to look at the unit test for the navigation and try to see what I can add.

File: Core/include/Acts/Propagator/Navigator.hpp [hunk 1]
```diff
@@ -622,9 +613,12 @@ class Navigator {
                                   << surface->geometryId());
       // Estimate the surface status
       bool boundaryCheck = true;
-      if (std::find(externalSurfaces.begin(), externalSurfaces.end(),
-                    surface->geometryId()) != externalSurfaces.end()) {
-        boundaryCheck = false;
```
Comment (asalzburger) [inline Core/include/Acts/Propagator/Navigator.hpp:627]: I do not understand why the search did not work but the loop over the external surfaces does ... can you explain, please?

**Review by asalzburger — COMMENTED** [review]
Can you please elaborate on my question?

File: Core/include/Acts/Propagator/Navigator.hpp (hunk 1 above)
Comment (Corentin-Allaire) [reply Core/include/Acts/Propagator/Navigator.hpp:627]: If I remember correctly, for this change both worked. This change is just to make it faster, I think the reasoning I had was that  since `state.navigation.externalSurfaces` is a multimap I couldn't use find with it directly so I had to fill a vector then used `std::find` with the vector. In the new version I just do a loop over the range so I don't have to do the copy.

**Comment by Corentin-Allaire** [comment]
I added some test for navigation in bounding volume hierarchy (no layer like the ATLAS Calo).

File: Core/include/Acts/Propagator/Navigator.hpp (hunk 1 above)
Comment (asalzburger) [reply Core/include/Acts/Propagator/Navigator.hpp:627]: ok, fine with me then.

Comment (Corentin-Allaire) [reply Core/include/Acts/Propagator/Navigator.hpp:627]: Sorry I missed your question. Yes that is the idea, to be honest the `externalSurfaces` were already there as a multimap just not implemented but I found it convenient that way so I kept them as is.

**Review by asalzburger — APPROVED** [review]

**Comment by asalzburger** [comment]
@paulgessinger I am approving & merging this one now, so feel free to build after merge.
